std/typecons.d

```diff
@@ -2522,6 +2527,12 @@ Returns:
         return _value;
     }
 
+    /// ditto
+    @property get(T datum) @safe pure nothrow
```

Comment:
This could be `const` as well?

You could get around the impure error by templating get and thus letting the compiler do the work.